scour/scour.py

```diff
@@ -1543,35 +1543,34 @@ def removeDuplicateGradients(doc):
                 continue
             master = bucket[0]
             duplicates = bucket[1:]
+            duplicates_ids = [d.getAttribute('id') for d in duplicates]
             master_id = master.getAttribute('id')
             if not master_id:
                 # If our selected "master" copy does not have an ID,
                 # then replace it with one that does (assuming any of
                 # them has one).  This avoids broken images like we
                 # saw in GH#203
-                for i in range(len(duplicates)):
-                    dup = duplicates[i]
-                    dup_id = dup.getAttribute('id')
+                for i in range(len(duplicates_ids)):
+                    dup_id = duplicates_ids[i]
                     if dup_id:
+                        # We do not bother updating the master field
+                        # as it is not used any more.
+                        master_id = duplicates_ids[i]
                         duplicates[i] = master
-                        master = dup
+                        duplicates_ids[i] = master_id
```

Comment:
Wait a moment - do I see correctly that we assign like
```py
master_id = duplicates_ids[i]
duplicates_ids[i] = master_id
```
i.e. effectively `duplicates_ids[i] = duplicates_ids[i]`?

This can go, right? We only need the line `duplicates[i] = master`

Reply:
Thanks for finding that.

Partly, we also "need" to clear `duplicates_ids[i]` to avoid remapping the master id to itself.  The end result is the same, but we would avoid some unnecessary remappings.

Reply:
Done.